File: src/com/company/superclass/LabWork.java
package com.company.superclass;

import com.company.Difficulty;


public interface LabWork {
    int getId();
    void setID(int id);
    String getName();
    void setName(String name);
    Coordinates getCoordinates();
    void setCoordinates(Coordinates coordinates);
    java.util.Date getCreationDate();
    void setCreationDate(java.util.Date creationDate);
    Integer getMinimalPoint();
    void setMinimalPoint(Integer minimalPoint);
    double getMaximumPoint();
    void setMaximumPoint(double maximumPoint);
    Difficulty getDifficulty();
    void setDifficulty(Difficulty difficulty);
    Discipline getDiscipline();
    void setDiscipline(Discipline discipline);
    String getLabWork(Boolean csv);
}

File: src/main/kotlin/com/company/collection/LabWork.kt
package com.company.collection

import com.company.superclasses.Difficulty
import java.util.Date

data class LabWork(val id: Int, val name: String, val coordinates: Coordinates,
                   val creationDate: Date, val minimalPoint: Int,
                   val maximalPoint: Double, val difficulty: Difficulty,
                   val discipline: Discipline) /*: Comparable<LabWork> {
    /**
     * LabWork-to-LabWork comparator
     *
     * @param other LabWork
     * @return >0 if this LabWork have more maximalPoint, <0 if LabWork have less maximalPoint, 0 if LabWork are of equal maximalPoint
     */
    override fun compareTo(other: LabWork): Int {
        return (this.maximalPoint - other.maximalPoint).toInt()
    }

}*/
